query-graphs-js/src/querygraph.ts

```diff
@@ -189,10 +197,15 @@ export class Edge {
   }
 
   addToConditions(newConditions: SelectionSet) {
-    if (!this._conditions) {
-      this._conditions = new SelectionSet(this.head.type as CompositeType);
-    }
+    // As mentioned in the ctor, we freeze the conditions to avoid unexpected modifications once a query
+    // graph has bee fully built (this method is called _during_ the building, so can still mutate the
+    // edge freely). Which means we need to clone any existing conditions (so we can modify them), and need
+    // to re-freeze the result afterwards.
+    this._conditions = this._conditions
+      ? this._conditions.clone()
```

Comment:
Is this clone() necessary? Aren't we already cloning in the constructor?

Reply:
We're also freezing in the constructor, and here we're trying to mutate the conditions to add new ones. So that `clone` is essentially about "unfreezing" the selections (but in theory, we could add an `unfreeze` method that just flip back the `_isFrozen` boolean, and that would be enough for that one specific case; but an `unfreeze` method makes the frozen story a bit dodgy and I don't think cloning here matters performance wise so ...).